src/s390_ecc.c

```diff
@@ -767,8 +781,18 @@ unsigned int ecdh_hw(ica_adapter_handle_t adapter_handle,
 
 	rc = ioctl(adapter_handle, ZSECSENDCPRB, xcrb);
 	if (rc != 0) {
-		rc = EIO;
-		goto ret;
+		dom_addressing = dom_addressing_default_domain;
+		reply_p = make_ecdh_request(privkey_A, pubkey_B, &xcrb, &buf, &len);
+		if (!reply_p) {
+			rc = EIO;
+			goto ret;
+		}
+
+		rc = ioctl(adapter_handle, ZSECSENDCPRB, xcrb);
+		if (rc != 0) {
+			rc = EIO;
+			goto ret;
+		}
```

Comment:
this duplicates the code ... and uses dom_addressing as a side effect.
what about 

typedef enum {
	dom_addressing_autoselect = 0,
	dom_addressing_default_domain = 1,
} dom_addressing_t;

...

for (domaddr = dom_addressing_autoselect; domaddr <= dom_addressing_default_domain; domaddr++) {
  ...
}
if (domaddr >dom_addressing_default_domain) {
  // both did not work ...
}

still with side-effect but the code is just once.

Btw. what's with reply_p pointing to ? Your code gives me the impression, it is not freed when the first attempt fails.